android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -486,18 +486,21 @@ protected static boolean removeLexicalModel(Context context, int position) {
    * @param context
    * @param position - int position in the models list
    */
-  protected static void deleteLexicalModel(Context context, int position) {
+  protected static void deleteLexicalModel(Context context, int position, boolean silenceNotification) {
     String modelID = getModelIDFromPosition(context, position);
     boolean result = removeLexicalModel(context, position);
 
     if (result) {
-      Toast.makeText(context, "Model deleted", Toast.LENGTH_SHORT).show();
+      if(!silenceNotification) {
+        Toast.makeText(context, "" + "Model deleted", Toast.LENGTH_SHORT).show();
```

Comment:
Looking good apart from using the term "Model" where we should be saying "Dictionary" in the UI :)

Do we have any resource strings in the Android UI? If so, is it possible for us to avoid adding extra UI text into the code which will make future localisation more tedious?